src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -235,11 +235,14 @@ export class FocusMonitor implements OnDestroy {
     // the shadow root, rather than the `document`, because the browser won't emit focus events
     // to the `document`, if focus is moving within the same shadow root.
     const rootNode = (_getShadowRoot(nativeElement) as HTMLElement|null) || this._getDocument();
+    const cachedInfo = this._elementInfo.get(nativeElement);
 
     // Check if we're already monitoring this element.
-    if (this._elementInfo.has(nativeElement)) {
-      const cachedInfo = this._elementInfo.get(nativeElement)!;
-      cachedInfo.checkChildren = checkChildren;
+    if (cachedInfo) {
+      if (checkChildren) {
+        cachedInfo.checkChildren = true;
```

Comment:
@mmalerba I think that this will work for most of the cases, but I'm not sure whether it's ideal. This means that the initial subscriber that signed up only for focus events on the element will now get emissions from children as well. Ideally, we'd be keeping track of two separate streams: own focus and child focus, but I didn't do it in this PR, because there are some issues with having two streams:

1. There can be a conflict between the CSS focus classes, because both the `checkChildren` and the non-`checkChildren` calls are trying to set the same classes on the same element.
2. We'd need a way to match focus either to the own focus stream or the child focus stream. Matching it is easy, but it can have performance implications, because we'd have to loop through and check all of the monitored elements on each `focus` and `blur` event.
3. We'd have to introduce a `checkChildren` parameter on `stopMonitoring` so that the two streams can be stopped independently. This would either have to be breaking API change, or we'd introduce memory leaks in people's code, because their `stopMonitoring` calls won't work anymore.

Reply:
Ok those are some good points, especially #⁠1. Can you file a design issue with the above and link it in a TODO? This PR seems like the best option for the moment, but it would be good to have that documented